--- includes/locale.inc	2011-07-28 00:02:24.000000000 +0300
+++ includes/locale.inc.new	2011-11-26 18:39:34.162878000 +0200
@@ -1710,6 +1710,7 @@
       'group' => 'all',
       'language' => 'all',
       'string' => '',
+      'exact_match' => 0,
     );
   }
 
@@ -1717,30 +1718,58 @@
   $sql_query->leftJoin('locales_target', 't', 't.lid = s.lid');
   $sql_query->fields('s', array('source', 'location', 'context', 'lid', 'textgroup'));
   $sql_query->fields('t', array('translation', 'language'));
-
-  // Compute LIKE section.
-  switch ($query['translation']) {
-    case 'translated':
-      $sql_query->condition('t.translation', '%' . db_like($query['string']) . '%', 'LIKE');
-      $sql_query->orderBy('t.translation', 'DESC');
-      break;
-    case 'untranslated':
-      $sql_query->condition(db_and()
-        ->condition('s.source', '%' . db_like($query['string']) . '%', 'LIKE')
-        ->isNull('t.translation')
-      );
-      $sql_query->orderBy('s.source');
-      break;
-    case 'all' :
-    default:
-      $condition = db_or()
-        ->condition('s.source', '%' . db_like($query['string']) . '%', 'LIKE');
-      if ($query['language'] != 'en') {
-        // Only search in translations if the language is not forced to English.
-        $condition->condition('t.translation', '%' . db_like($query['string']) . '%', 'LIKE');
+  
+  if(!$query['exact_match']) {
+      // Compute LIKE section.
+      switch ($query['translation']) {
+        case 'translated':
+          $sql_query->condition('t.translation', '%' . db_like($query['string']) . '%', 'LIKE');
+          $sql_query->orderBy('t.translation', 'DESC');
+          break;
+        case 'untranslated':
+          $sql_query->condition(db_and()
+            ->condition('s.source', '%' . db_like($query['string']) . '%', 'LIKE')
+            ->isNull('t.translation')
+          );
+          $sql_query->orderBy('s.source');
+          break;
+        case 'all' :
+        default:
+          $condition = db_or()
+            ->condition('s.source', '%' . db_like($query['string']) . '%', 'LIKE');
+          if ($query['language'] != 'en') {
+            // Only search in translations if the language is not forced to English.
+            $condition->condition('t.translation', '%' . db_like($query['string']) . '%', 'LIKE');
+          }
+          $sql_query->condition($condition);
+          break;
+      }
+  }
+  else {
+      // Compute EXACT MATCH section.
+      switch ($query['translation']) {
+        case 'translated':
+          $sql_query->condition('t.translation', $query['string']);
+          $sql_query->orderBy('t.translation', 'DESC');
+          break;
+        case 'untranslated':
+          $sql_query->condition(db_and()
+            ->condition('s.source', $query['string'])
+            ->isNull('t.translation')
+          );
+          $sql_query->orderBy('s.source');
+          break;
+        case 'all' :
+        default:
+          $condition = db_or()
+            ->condition('s.source',$query['string']);
+          if ($query['language'] != 'en') {
+            // Only search in translations if the language is not forced to English.
+            $condition->condition('t.translation', $query['string']);
+          }
+          $sql_query->condition($condition);
+          break;
       }
-      $sql_query->condition($condition);
-      break;
   }
 
   $limit_language = NULL;
@@ -1801,7 +1830,7 @@
   $query = &drupal_static(__FUNCTION__);
   if (!isset($query)) {
     $query = array();
-    $fields = array('string', 'language', 'translation', 'group');
+    $fields = array('string', 'language', 'translation', 'group', 'exact_match');
     foreach ($fields as $field) {
       if (isset($_SESSION['locale_translation_filter'][$field])) {
         $query[$field] = $_SESSION['locale_translation_filter'][$field];
--- modules/locale/locale.admin.inc	2011-07-28 00:02:24.000000000 +0300
+++ modules/locale/locale.admin.inc.new	2011-11-26 18:23:19.712601000 +0200
@@ -813,9 +813,13 @@
   unset($languages['en']);
 
   $filters['string'] = array(
-    'title' => t('String contains'),
+    'title' => t('String'),
     'description' => t('Leave blank to show all strings. The search is case sensitive.'),
   );
+  $filters['exact_match'] = array(
+    'title' => t('Exact match'),
+    'description' => t('Check to do an exact match.'),
+  );
 
   $filters['language'] = array(
     'title' => t('Language'),
@@ -859,6 +863,13 @@
         '#description' => $filter['description'],
       );
     }
+    else if ($key == 'exact_match') {
+      $form['filters']['status']['exact_match'] = array(
+        '#type' => 'checkbox',
+        '#title' => $filter['title'],
+        '#description' => $filter['description'],
+      );
+    }
     else {
       $form['filters']['status'][$key] = array(
         '#title' => $filter['title'],
